test/model/src/include/model/driver/kv_workload_runner.h

```diff
@@ -78,7 +78,7 @@ class kv_workload_runner {
                * point we would record and compare return codes. But we're not there yet, so just
                * fail on error.
                */
-              if (ret != 0)
+              if (ret != 0 && ret != WT_NOTFOUND)
```

Comment:
I added this so that I can generate `remove` operations without having to worry about whether the key exists. I think of "not found" more of as an expected behavior, as opposed to more genuine errors.

Normal "lightweight verification" would record all of these return codes and compare them across WiredTiger and model executions, but we're not there yet—this is something that should be added in the future. (I don't want to do this now, because I'd have to redo it anyways after we get to handle crashes and multi-threaded execution, as they both have implications about how recording return codes would work.)

But in our case, we still end up comparing the database at the end between WiredTiger and the model. If there is any inconsistency in remove between the two, it will be caught by that comparison with a high probability, and tests will be executed with many repetitions—so for all practical purposes, it should be safe. And then we'll make it better once we get to implement recording and comparing return values.

Reply:
Got it - thank you for explaining that.

Reply:
That's a good question. At this point, the expectation is that with enough workload executions, a difference in behavior between WiredTiger and the model would be uncovered. But I agree that especially for subtle bugs, this may not happen—or it may not happen in a while.

My plan is to still record the return codes and compare them across WiredTiger and model executions in the future, but I wanted to work out a few more things before that point. I created WT-12359 to track this.